### DISCUSSION TITLE
formState.name and typescript error

### DISCUSSION THREAD

**Discussion by npearson72** [react-hook-form/react-hook-form#9308] [Q&A] [answered]
First off, thank you for the amazing library!

When I log the `formState` object that I get from the `useForm` hook, it appears to have a `name` field.

<img width="295" alt="Screen Shot 2022-11-01 at 6 20 11 PM" src="https://user-images.githubusercontent.com/1323414/199372456-ab231346-7d41-4ed3-9a90-2ca2de1bddce.png">

This `name` field appears to indicate the "current" input that was changed (or interacted with entirely... not totally sure).

But when I try to use it I'm getting a typescript error:

<img width="387" alt="Screen Shot 2022-11-01 at 6 23 23 PM" src="https://user-images.githubusercontent.com/1323414/199372734-a3618a93-6943-4cfe-91f4-c9fb8b74aa86.png">

Is there a way to resolve the typescript error?

Or, is there another way to get the "current" input? I'm trying to tigger a different field's validations while using the `Controller`.

**Comment by bluebill1049** [comment, marked as answer]
It shouldn't have a name in formState or else this is a bug.

```tsx
export type FormState<TFieldValues extends FieldValues> = {
  isDirty: boolean;
  isSubmitted: boolean;
  isSubmitSuccessful: boolean;
  isSubmitting: boolean;
  isValidating: boolean;
  isValid: boolean;
  submitCount: number;
  defaultValues?: Readonly<DeepPartial<TFieldValues>> | TFieldValues;
  dirtyFields: Partial<Readonly<FieldNamesMarkedBoolean<TFieldValues>>>;
  touchedFields: Partial<Readonly<FieldNamesMarkedBoolean<TFieldValues>>>;
  errors: FieldErrors<TFieldValues>;
};
```

**Reply by npearson72** [reply]
It does indeed. It gets added on blur. If this is an unreliable field, is there any other way to know the current / most recently blurred field at the formState level? I'm trying to check on this in a useEffect.

Thanks.

**Reply by bluebill1049** [reply]
can you share a reproducible csb for this added `name`?

![Screenshot 2022-11-03 at 4 31 05 pm](https://user-images.githubusercontent.com/10513364/199653454-1dadac5b-fdf8-452d-85a4-c5596a55d333.png)

**Reply by npearson72** [reply]
Yes. Will create one tomorrow and share it with you. Thanks for responding.

**Reply by npearson72** [reply]
You know, actually... I found a different (probably better) way of handling my case:

```
        <Controller
          render={({ field, fieldState, formState }) => {
            const handleOnChange = (value: string) => {
              field.onChange(value);

              if (formState.touchedFields.confirmPassword) {
                trigger('confirmPassword');
              }
            };
            return (
              <CustomPasswordInput
                label={t('label.password')}
                controlledValue={field.value}
                setValueOnChange={handleOnChange}
                setValueOnBlur={field.onBlur}
                error={fieldState.error?.message}
              />
            );
          }}
          name="password"
          control={control}
        />
```

I didn't know I could get the `formState` off of the render callback in the `Controller` component, so I was using `formState` off of the `UseFormReturn` object within a `useEffect` watcher. In that case, I needed to know which field was the "current" field that the user was interacting with and I found that a dynamic `name` key would appear. I realize it's not on the `FormState` type -- that's why I was asking about it.

But it seems like it's probably not a reliable field to use, and in general, this new approach I've discovered appears to be more idiomatic react-hook-form.

Long-story-short, I'll probably not be creating a csb for lack of time.

Thanks again for your help.
